test/files/run/t11396.scala

```diff
@@ -0,0 +1,664 @@
+
+object Test extends App {
+  import scala.collection.immutable.Queue
+  val three = Queue(1, 2, 3)
+  val six   = three.enqueueAll(4 to 6)
+  val it    = six.iterator
+  val _     = six.enqueue(7)
+  assertEquals(1, it.next())
+
+  def assertEquals(x: Int, y: Int) = assert(x == y)
+}
+
+/* With .javaopts, test is forked with test classes ahead of standard library on the class path.
+ * This is a copy of List at the time of writing, with reverse rigged to blow, like those movies
```

Comment:
I feel like there has to be a way to do this which doesn't involve cloning the class (which is a pain to keep up-to-date). Perhaps using reflection to replace `Queue#in` with a `Proxy` and `InvocationHandler`?

Reply:
The library is locked down until 3.2, so I went for quick and dirty. I considered minimizing the clone to what's required for the test, which is doable.